### ISSUE TITLE
Enable RH Registry signature verification by default

### ISSUE THREAD

**Issue by jasinner** [openshift/machine-config-operator#1349]
**Description**

All images in the Red Hat registry are signed. We have the GPG key on disk to enable verification of these images when pulling, all that's left to do is configure Cri-o to verify the signatures when pulling those images.

**Steps to reproduce the issue:**

See: https://access.redhat.com/verify-images-ocp4

**Describe the results you received:**

Signatures are not verified

**Describe the results you expected:**

Verification steps from the reproduction article are met

**Output of `oc adm release info --commits | grep machine-config-operator`:**

```
machine-config-operator                       https://github.com/openshift/machine-config-operator                       d780d197a9c5848ba786982c0c4aaa7487297046
```

**Additional environment details (platform, options, etc.):**

OCP 4.2.12

**Comment by smarterclayton** [comment]
Are there any negative implications of turning on verification?  Would it fail for images that don’t have signatures for some reason (and are we certain all of them have signatures)?

**Comment by cgwalters** [comment]
This almost may merit [an enhancement](https://github.com/openshift/enhancements/) but I'd also be fine just trying it out in master (4.4) for a bit...

**Comment by adambkaplan** [comment]
Would we want builds to perform similar signature checks for the base/builder images?

**Comment by jasinner** [comment]
> Are there any negative implications of turning on verification? Would it fail for images that don’t have signatures for some reason (and are we certain all of them have signatures)?

According to [DELIVERY-5889](https://projects.engineering.redhat.com/browse/DELIVERY-5889) and [DELIVERY-6699](https://projects.engineering.redhat.com/browse/DELIVERY-6699) they should all be signed. If an image doesn't have a signature and your pulling from one of the Red Hat repos (registry.access.redhat.com), or (registry.redhat.io) the pull will fail.

**Comment by jasinner** [comment]
> Would we want builds to perform similar signature checks for the base/builder images?

Sure, would not.

**Comment by adambkaplan** [comment]
> This almost may merit an enhancement but I'd also be fine just trying it out in master (4.4) for a bit...

@cgwalters I'd like to see an enhancement so we can discuss what to do in builds. It's unclear to me how frequently customers use non-Red Hat images in their build pipelines.

**Comment by bparees** [comment]
> Are there any negative implications of turning on verification? Would it fail for images that don’t have signatures for some reason (and are we certain all of them have signatures)?

just for posterity:  today the catalog images consumed by OLM are not signed.  We're working on fixing that, though it looks like they'll be signed w/ a different key than the standard release key.  So any effort to enable signature verification by default needs to take into account how the olm catalog images can be verified (or exempt them from verification)

**Comment by JAORMX** [comment]
> > Are there any negative implications of turning on verification? Would it fail for images that don’t have signatures for some reason (and are we certain all of them have signatures)?
> 
> just for posterity: today the catalog images consumed by OLM are not signed. We're working on fixing that, though it looks like they'll be signed w/ a different key than the standard release key. So any effort to enable signature verification by default needs to take into account how the olm catalog images can be verified (or exempt them from verification)

Is this still the case? Are even redhat-provided catalog images unsigned?

**Comment by bparees** [comment]
> Is this still the case? Are even redhat-provided catalog images unsigned?

yes.  should be resolved soon.  Here's some more info on how you can enable verification for everything else (the content the catalog references is still signed):

https://bugzilla.redhat.com/show_bug.cgi?id=1903632#c38

**Comment by cgwalters** [comment]
I do think we should do this by default when we're ready.
/reopen
/lifecycle frozen
